torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1488,7 +1488,7 @@ void InitXlaModuleBindings(py::module m) {
         std::vector<XLATensor::ShardingSpecPtr>{sharding_spec},
         std::vector<std::string>{GetVirtualDevice().toString()})[0];
     xtensor->SetXlaData(xla_data);
-    xtensor->SetShardingSpec(*sharding_spec);
+    xtensor->CreateShardedIrValue(*sharding_spec);
```

Comment:
Let's also do e2e training test, and verify that this doesn't affect the compilation & caching to be cold throughout the loop.